kong/plugins/prometheus/handler.lua

```diff
@@ -34,6 +37,26 @@ function PrometheusHandler.log(self, conf)
     end
   end
 
+  local tags = {}
+  if conf.expose_tags.from_service and message.service and message.service.tags then
+    for i = 1, #message.service.tags do
+      tags[message.service.tags[i]] = true
+    end
+  end
+  if conf.expose_tags.from_route and message.route and message.route.tags then
+    for i = 1, #message.route.tags do
+      tags[message.route.tags[i]] = true
+    end
+  end
+  if conf.expose_tags.from_consumer and message.consumer and message.consumer.tags then
+    for i = 1, #message.consumer.tags do
+      tags[message.consumer.tags[i]] = true
+    end
+  end
+  tags = keys(tags)
+  sort(tags)
+  serialized.tags = concat(tags, ",")
+
```

Comment:
This plugin has already faced a lot of performance issues under high request volume and high cardinality of services/routes.
I'm a bit concerned about the performance impact of this change.
We will need to measure performance before we hit the merge button.